Doppler.HtmlEditorApi/Controllers/CampaignsController.cs

```diff
@@ -143,22 +143,39 @@ public async Task<IActionResult> SaveCampaign(string accountName, int campaignId
                 _ => throw new NotImplementedException($"Unsupported campaign content type {campaignContent.type:G}")
             };
 
-            if (campaignState.ContentExists)
+            var campaignIds = campaignState.TestABCondition == TestABCondition.TypeClassic
+            ? new int[] { campaignState.IdCampaignA }
+            : new int[] { campaignState.IdCampaignA, campaignState.IdCampaignB };
+
+            foreach (var campaign in campaignIds)
             {
-                await _campaignContentRepository.UpdateCampaignContent(accountName, contentRow);
+                if (campaignState.ContentExists)
+                {
+                    await _campaignContentRepository.UpdateCampaignContent(campaign, contentRow);
+                }
+                else
+                {
+                    await _campaignContentRepository.CreateCampaignContent(campaign, contentRow);
+                    await _campaignContentRepository.UpdateCampaignStatus(
+                        setCurrentStep: 2,
+                        setHtmlSourceType: TemplateHtmlSourceType,
+                        whenIdCampaignIs: campaign,
+                        whenCurrentStepIs: 1);
+                }
+                await _campaignContentRepository.UpdateCampaignPreviewImage(campaign, contentRow.PreviewImage);
+                await _campaignContentRepository.SaveNewFieldIds(campaign, fieldIds);
+                await _campaignContentRepository.SaveLinks(campaign, trackableUrls);
             }
-            else
+
+            if (campaignState.TestABCondition == TestABCondition.TypeTestABSubject)
```

Comment:
What do you think about using `IdCampaignResult` in place of `TestABCondition`?  

```suggestion
            if (campaignState.IdCampaignResult != null)
```

Reply:
Good!